### PR TITLE
[ee-TS] LPS-110420 searchEngineId can't be configured DLFileEntry search / index operations

### PR DESCRIPTION
**Pull request by joshuacords**
https://issues.liferay.com/browse/LPS-110420
https://issues.liferay.com/browse/LPP-36740

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshuacords** [comment]
ci:test:search-solr

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/engine/SearchEngineIdProviderImpl.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+		if (Validator.isNotNull(searchEngineId)) {
+			_searchEngineId = searchEngineId;
+		}
+		else {
+			_searchEngineId = SearchEngineHelper.SYSTEM_ENGINE_ID;
+		}
+	}
+
+	private String _searchEngineId = SearchEngineHelper.SYSTEM_ENGINE_ID;
```
Comment (lipusz) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/engine/SearchEngineIdProviderImpl.java:62]: Couldn't this be coming from `SearchEngineIdConfiguration` by defining a default value for the `indexSearchEngineId` property? @joshuacords

Comment (joshuacords) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/engine/SearchEngineIdProviderImpl.java:62]: Yes, but I was advised by Andre to use this approach:

> Suggestion: remove deflt from SearchEngineIdConfiguration, meaning when Admins see it empty they’ll think “DXP decides unless I override”. (kudos to Minhchau for first articulating this school of thought)

Comment (lipusz) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/engine/SearchEngineIdProviderImpl.java:62]: I've never heard about this "approach" before... Empty is usually not a "reasonable default" when it has hidden meaning. :-)

Comment (joshuacords) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/engine/SearchEngineIdProviderImpl.java:62]: Shrug - I'm not sure. If you want to push on it loop Andre in.

Comment (lipusz) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/engine/SearchEngineIdProviderImpl.java:62]: At least a value which is there and visible does not leave you in question. The interpretation of empty, as you can see, can be different on the other hand.

**Pull request closed without merging**